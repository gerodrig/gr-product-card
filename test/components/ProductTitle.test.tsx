import React from 'react';
import renderer from 'react-test-renderer';
import { ProductTitle, ProductCard } from '../../src/components'
import { product1 } from '../data/products';


describe('tests in ProductTitle component', () => {
    
    test('should render ProductTitle component with the custom title', () => {
        const wrapper = renderer.create(<ProductTitle title="Custom Product" />);

        expect(wrapper.toJSON()).toMatchSnapshot();
    }
    );

    test('should show the title as product name if title is not defined', () => {
        const wrapper = renderer.create(
            <ProductCard product={product1}>
                {
                    () => (
                        <ProductTitle />
                    )
                }
            </ProductCard>
        );

        expect(wrapper.toJSON()).toMatchSnapshot();
    })
    
})
