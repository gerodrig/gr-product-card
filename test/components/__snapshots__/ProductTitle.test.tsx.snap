// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests in ProductTitle component should render ProductTitle component with the custom title 1`] = `
<span
  className="productDescription undefined"
>
  Custom Product
</span>
`;

exports[`tests in ProductTitle component should show the title as product name if title is not defined 1`] = `
<div
  className="productCard undefined"
>
  <span
    className="productDescription undefined"
  >
    Coffe Mug - No Image!
  </span>
</div>
`;
